### ISSUE TITLE
Unable to open archive.info.

### ISSUE THREAD

**Issue by DarylAndDaryl** [pgbackrest/pgbackrest#2008]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: **2.44**

2. PostgreSQL version: **13.7**

3. Operating system/version: **Red Hat Enterprise Linux 8.6**

4. Did you install pgBackRest from source or from a package? **package**

5. Please attach the following as applicable:
    **_pgbackrest.conf_** (location: /etc/pgbackrest.conf): 
        
        [global]
        repo1-path=/pgsql/rubrik
        repo1-host-user=postgres
        repo1-retention-full=1
        process-max=2
        log-level-console=info
        log-level-file=debug
        start-fast=y

        [PgSqlBackup]
        pg1-path=/var/lib/pgsql/data

    **_postgresql.conf`_**:

         archive_command = 'pgbackrest --stanza=PgSqlBackup archive-push %p'
         archive_mode = on
         listen_addresses = '*'
         #max_wal_senders = 10
         #wal_level = replica
         #port = 5432

    **_errors in the postgresql log file_**:

           ERROR: [103]: unable to find a valid repository:
           repo1: [FileOpenError] unable to load info file '/pgsql/rubrik/archive/PgSqlBackup/archive.info' or 
           '/pgsql/rubrik/archive/PgSqlBackup/archive.info.copy':
           FileOpenError: unable to open file '/pgsql/rubrik/archive/PgSqlBackup/archive.info' for read: [13] Permission denied
           FileOpenError: unable to open file '/pgsql/rubrik/archive/PgSqlBackup/archive.info.copy' for read: [13] Permission denied
           HINT: archive.info cannot be opened but is required to push/get WAL segments.
           HINT: is archive_command configured correctly in postgresql.conf?
           HINT: has a stanza-create been performed?
           HINT: use --no-archive-check to disable archive checks during backup if you have an alternate archiving scheme.
           2023-02-22 14:02:08.302 P00   INFO: archive-push command end: aborted with exception [103]

7. Describe the issue:

- pgbackrest --stanza=PgSqlBackup stanza-create: successful
- pgbackrest --stanza=PgSqlBackup check: fails

I issue the command manually from a command prompt and under the Linux account that owns PostgreSQL and pgBackRest, which is postgres.  The password is locked but I get the same issue when the password is not locked.

I made sure the directory paths are owned (rwx) by postgres.

The archive file exist because the stanza-create statement created it with an owner of postgres with rw permissions.

If I changed the repo1-path to a folder with the postgres home path, then everything works perfectly but the goal is to ultimately use an NFS share mounted outside of the postgres home directory.

In summary, I sudo to the postgres account. I issue the pgbackrest stanza-create, which builds out the "archive" and "backup" folders within the repo1-path.  The pgbackrest check fails with permission denied when trying to read the archive.info file.

Any help solving this would be greatly appreciated.

Thank you.

**Comment by pgstef** [comment]
Hi,

> If I changed the repo1-path to a folder with the postgres home path, then everything works perfectly but the goal is to ultimately use an NFS share mounted outside of the postgres home directory.

Unfortunately, it really sounds like a system restriction.

I've got a test cluster running on `generic/rhel8` vagrant boxes and it seems to work fine:

```bash
$ cat /etc/redhat-release 
Red Hat Enterprise Linux release 8.7 (Ootpa)
$ cat /etc/fstab |grep backups
srv2:/mnt/backups /mnt/backups nfs rw,sync,hard,intr 0 0
$ mount |grep /mnt/backup
srv2:/mnt/backups on /mnt/backups type nfs4 
	(rw,relatime,sync,vers=4.2,rsize=524288,wsize=524288,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,...)

$ sudo -iu postgres pgbackrest --stanza=demo info
stanza: demo
    status: ok
    cipher: aes-256-cbc

    db (current)
        wal archive min/max (14): 000000010000000000000004/000000010000000000000004

        full backup: 20230224-092946F
            timestamp start/stop: 2023-02-24 09:29:46 / 2023-02-24 09:30:02
            wal start/stop: 000000010000000000000004 / 000000010000000000000004
            database size: 56.8MB, database backup size: 56.8MB
            repo1: backup set size: 9.2MB, backup size: 9.2MB

$ sudo ls -l /mnt/
total 0
drwxr-xr-x. 4 postgres postgres 35 Feb 24 09:26 backups
$ sudo ls -l /mnt/backups/
total 0
drwxr-x---. 3 postgres postgres 18 Feb 24 09:26 archive
drwxr-x---. 3 postgres postgres 18 Feb 24 09:26 backup
$ sudo ls -l /mnt/backups/archive/
total 0
drwxr-x---. 3 postgres postgres 63 Feb 24 09:30 demo
$ sudo ls -l /mnt/backups/archive/demo/
total 8
drwxr-x---. 3 postgres postgres  30 Feb 24 09:27 14-1
-rw-r-----. 1 postgres postgres 368 Feb 24 09:30 archive.info
-rw-r-----. 1 postgres postgres 368 Feb 24 09:30 archive.info.copy
```

So it's probably a system configuration (maybe selinux) preventing your PG server to access files outside its user directories?

Kind Regards

**Comment by DarylAndDaryl** [comment]
Thank you for Stefan for the info.  I will work with my local Linux administrator to see if we can determine the issue.

I see you indicated the version of RHEL you used is version 8.7, which mine is 8.6.  Since your pgbackrest created a 14-1 file, I believe that means your PosgreSQL version is 14, where mine is 13.7.  Would you please provide the version of pgbackrest you used?  Mine is 2.44.

Sincerely,


Daryl Williams
(he/him/his)
InfraOps Database Administration Team Lead
Delta Dental of Michigan, Ohio, and Indiana
Ph: 517-347-5853
***@***.***
Certified Call Center of Excellence  •  Rated A (Excellent) by A.M. Best Company

From: Stefan Fercot ***@***.***>
Sent: Friday, February 24, 2023 4:50 AM
To: pgbackrest/pgbackrest ***@***.***>
Cc: Daryl Williams ***@***.***>; Author ***@***.***>
Subject: [External] Re: [pgbackrest/pgbackrest] Unable to open archive.info. (Issue #2008)


Hi,

If I changed the repo1-path to a folder with the postgres home path, then everything works perfectly but the goal is to ultimately use an NFS share mounted outside of the postgres home directory.

Unfortunately, it really sounds like a system restriction.

I've got a test cluster running on generic/rhel8 vagrant boxes and it seems to work fine:

$ cat /etc/redhat-release

Red Hat Enterprise Linux release 8.7 (Ootpa)

$ cat /etc/fstab |grep backups

srv2:/mnt/backups /mnt/backups nfs rw,sync,hard,intr 0 0

$ mount |grep /mnt/backup

srv2:/mnt/backups on /mnt/backups type nfs4

 (rw,relatime,sync,vers=4.2,rsize=524288,wsize=524288,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,...)



$ sudo -iu postgres pgbackrest --stanza=demo info

stanza: demo

    status: ok

    cipher: aes-256-cbc



    db (current)

        wal archive min/max (14): 000000010000000000000004/000000010000000000000004



        full backup: 20230224-092946F

            timestamp start/stop: 2023-02-24 09:29:46 / 2023-02-24 09:30:02

            wal start/stop: 000000010000000000000004 / 000000010000000000000004

            database size: 56.8MB, database backup size: 56.8MB

            repo1: backup set size: 9.2MB, backup size: 9.2MB



$ sudo ls -l /mnt/

total 0

drwxr-xr-x. 4 postgres postgres 35 Feb 24 09:26 backups

$ sudo ls -l /mnt/backups/

total 0

drwxr-x---. 3 postgres postgres 18 Feb 24 09:26 archive

drwxr-x---. 3 postgres postgres 18 Feb 24 09:26 backup

$ sudo ls -l /mnt/backups/archive/

total 0

drwxr-x---. 3 postgres postgres 63 Feb 24 09:30 demo

$ sudo ls -l /mnt/backups/archive/demo/

total 8

drwxr-x---. 3 postgres postgres  30 Feb 24 09:27 14-1

-rw-r-----. 1 postgres postgres 368 Feb 24 09:30 archive.info

-rw-r-----. 1 postgres postgres 368 Feb 24 09:30 archive.info.copy

So it's probably a system configuration (maybe selinux) preventing your PG server to access files outside its user directories?

Kind Regards

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_pgbackrest_pgbackrest_issues_2008-23issuecomment-2D1443374162&d=DwMCaQ&c=Jnf2Cp4PGnCPtuPxt-OIQR21MChqGaHuqvEjP9X2AA0&r=tU36R6flun1OkNp3Dr7Y_0l3sc0oMsbS3thUP61qm9Q&m=E3jRiPImeebIHWJR2GaTz2XIoKMEBTlN62DeE6q3Ppkz-dnhPHajn3wAX1nD725e&s=IPSx0kAjGrHlwkZ18N0_cf24UTl6-C9P06PBvGe9XlI&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_A4HE7MHGOOFCIRSKTPHZNQDWZB73XANCNFSM6AAAAAAVEXVW5A&d=DwMCaQ&c=Jnf2Cp4PGnCPtuPxt-OIQR21MChqGaHuqvEjP9X2AA0&r=tU36R6flun1OkNp3Dr7Y_0l3sc0oMsbS3thUP61qm9Q&m=E3jRiPImeebIHWJR2GaTz2XIoKMEBTlN62DeE6q3Ppkz-dnhPHajn3wAX1nD725e&s=y2jNfKhnmc8sUyC2XKwVasmth8iDQbirETTXoeGnXts&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

**********************************************************************
CONFIDENTIALITY NOTICE: The information contained in this email is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged material. If you are not the intended recipient, you are hereby notified that any unauthorized review, use, dissemination, distribution or copying of this communication is prohibited and may be subject to legal restriction or sanction. If you have received this email in error, please notify the sender immediately to arrange for return or destruction of the information and all copies. If you are the intended recipient but do not wish to receive communications through this medium, please advise the sender immediately. Thank you.

**Comment by pgstef** [comment]
For the test purpose I used [this](https://app.vagrantup.com/generic/boxes/rhel8) vagrant box which indeed provides 8.7. I tried with both PG 14 and EDB Postgres Advanced Server 14 to see if there was any potential issue with the packages there.

Also as mentioned, I really don't see why it would be a pgBackRest problem if the PG server can't read files outside its own home directory. So I tried with the latest release available in the PGDG repository which is indeed 2.44.

I hope your Linux admin will be able to help you find out how the underlying server has been installed and configured 🤞 

NFS mounts can be tricky with ownerships. You could try with another repo-type to see if that's a postgres system user issue or an issue with your NFS share/mount point 🤔

**Comment by DarylAndDaryl** [comment]
Thanks Stefan for that info.

Please note, I did test with other directories that were not NFS shares.  For instance I tried utilizing a directory off /etc and a new mount point that was not NFS and both failed with the same error.  Note pgbackrest stanza-check works successfully in all cases.  It builds out the archive and backup directories and files within those directories.  Thus, the postgres Linux account can write to them including the NFS share.  I could manually use the postgres Linux account to read and write to all directories.  Its when I execute the pgbackrest check and try to do a full backup that I get the error of unable to open the archive.info file.

In summary, postgres can read and write to the NFS share and any other directory but it cannot read the archive.info file via pgbackrest.

Sincerely,


Daryl Williams
(he/him/his)
InfraOps Database Administration Team Lead
Delta Dental of Michigan, Ohio, and Indiana
Ph: 517-347-5853
***@***.***
Certified Call Center of Excellence  •  Rated A (Excellent) by A.M. Best Company

From: Stefan Fercot ***@***.***>
Sent: Friday, February 24, 2023 12:43 PM
To: pgbackrest/pgbackrest ***@***.***>
Cc: Daryl Williams ***@***.***>; Author ***@***.***>
Subject: [External] Re: [pgbackrest/pgbackrest] Unable to open archive.info. (Issue #2008)


For the test purpose I used this<https://urldefense.proofpoint.com/v2/url?u=https-3A__app.vagrantup.com_generic_boxes_rhel8&d=DwMFaQ&c=Jnf2Cp4PGnCPtuPxt-OIQR21MChqGaHuqvEjP9X2AA0&r=tU36R6flun1OkNp3Dr7Y_0l3sc0oMsbS3thUP61qm9Q&m=5mnXnuf-TwkHmoUwYh1cWDav9X7n3Flo2E3lCWH9gglfSNVZK8wUeG9c8UELUcLs&s=gsBbXk8GHJN8APb5r1kRMXv5japoKy4ca8cU5Q08z9s&e=> vagrant box which indeed provides 8.7. I tried with both PG 14 and EDB Postgres Advanced Server 14 to see if there was any potential issue with the packages there.

Also as mentioned, I really don't see why it would be a pgBackRest problem if the PG server can't read files outside its own home directory. So I tried with the latest release available in the PGDG repository which is indeed 2.44.

I hope your Linux admin will be able to help you find out how the underlying server has been installed and configured 🤞

NFS mounts can be tricky with ownerships. You could try with another repo-type to see if that's a postgres system user issue or an issue with your NFS share/mount point 🤔

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_pgbackrest_pgbackrest_issues_2008-23issuecomment-2D1444114561&d=DwMFaQ&c=Jnf2Cp4PGnCPtuPxt-OIQR21MChqGaHuqvEjP9X2AA0&r=tU36R6flun1OkNp3Dr7Y_0l3sc0oMsbS3thUP61qm9Q&m=5mnXnuf-TwkHmoUwYh1cWDav9X7n3Flo2E3lCWH9gglfSNVZK8wUeG9c8UELUcLs&s=bbsqvuDRfS0QcWeheV-WdOFJ-PyHcijcX0lPWqMEd00&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_A4HE7MHONYTTXNP4NQ6RO6DWZDXJTANCNFSM6AAAAAAVEXVW5A&d=DwMFaQ&c=Jnf2Cp4PGnCPtuPxt-OIQR21MChqGaHuqvEjP9X2AA0&r=tU36R6flun1OkNp3Dr7Y_0l3sc0oMsbS3thUP61qm9Q&m=5mnXnuf-TwkHmoUwYh1cWDav9X7n3Flo2E3lCWH9gglfSNVZK8wUeG9c8UELUcLs&s=UsFm2sC1Ejac-WkZbgBxldo0Wl1eGIBTDcs6RGh_V-c&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

**********************************************************************
CONFIDENTIALITY NOTICE: The information contained in this email is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged material. If you are not the intended recipient, you are hereby notified that any unauthorized review, use, dissemination, distribution or copying of this communication is prohibited and may be subject to legal restriction or sanction. If you have received this email in error, please notify the sender immediately to arrange for return or destruction of the information and all copies. If you are the intended recipient but do not wish to receive communications through this medium, please advise the sender immediately. Thank you.

**Comment by pgstef** [comment]
> In summary, postgres can read and write to the NFS share and any other directory but it cannot read the archive.info file via pgbackrest.

Not really. The postgres system user when logged with a tty maybe. It doesn't guarantee that the PG processes can.

I don't think it's a problem with the check or backup commands but more with archive-push which is called by the PG archiver process. 
It might worth to try using a shell command (a simple cp like in PG docs) to check that the archive_command can really write things there ? 🤔
